processing/src/main/java/org/apache/druid/query/aggregation/first/DoubleFirstAggregatorFactory.java

```diff
@@ -142,12 +160,24 @@ public VectorAggregator factorizeVector(
       VectorColumnSelectorFactory columnSelectorFactory
   )
   {
+    VectorValueSelector timeSelector = columnSelectorFactory.makeValueSelector(timeColumn);
     ColumnCapabilities capabilities = columnSelectorFactory.getColumnCapabilities(fieldName);
+
     if (Types.isNumeric(capabilities)) {
       VectorValueSelector valueSelector = columnSelectorFactory.makeValueSelector(fieldName);
-      VectorValueSelector timeSelector = columnSelectorFactory.makeValueSelector(
-          timeColumn);
-      return new DoubleFirstVectorAggregator(timeSelector, valueSelector);
+      VectorObjectSelector objectSelector = ExpressionVectorSelectors.castValueSelectorToObject(
+          columnSelectorFactory.getReadableVectorInspector(),
+          fieldName,
+          valueSelector,
+          capabilities.toColumnType(),
+          ColumnType.DOUBLE
+      );
+      return new DoubleFirstVectorAggregator(timeSelector, objectSelector);
+    }
+
+    VectorObjectSelector vSelector = columnSelectorFactory.makeObjectSelector(fieldName);
+    if (capabilities != null) {
+      return new DoubleFirstVectorAggregator(timeSelector, vSelector);
```

Comment:
Why is this check required? I don't think we are using the capabilities inside the aggregator